### PR TITLE
SDKNEW-3214 - E2E tests alignment

### PR DESCRIPTION
**Pull request by etrpx**
- Added RequestFilter to consolidate all request filters for improved management and organization.
- Added CustomFilterByFunction feature to enhance filtering capabilities.
- Fixed Context URI to Servlet Path for better compatibility.
- Fixed Cookie Issue with 2-Byte Encoded Characters**: Resolved an issue related to cookies containing 2-byte encoded characters. This problem occurred due to incorrect length handling in the input stream within the `buildBaseRequest` method in `PXApacheHttpClient.java`.
- Fixed Risk Request Schema to include the `client_uuid` for better data handling and analysis.
- Fixed the Block and Captcha pages, aligning them with the specified design and adding hard block functionality to align with spec.
- Fixed Risk UUID to ensure it is set even when encountering a server-to-server error.
- Fixed Async Activities Schema Addressed issues with the activity schema to ensure data accuracy and integrity.
- Aligned E2E tests

### CODE REVIEW

**Review by ori-gold-px — COMMENTED** [review]
A few comments but overall looks good

File: src/main/java/com/perimeterx/models/activities/CommonActivityDetails.java [hunk 1]
```diff
@@ -21,20 +21,32 @@ public class CommonActivityDetails implements ActivityDetails {
...
 
         if(loginData != null && loginData.getLoginCredentials() != null) {
             this.ciProtocol = loginData.getLoginCredentials().getCiProtocol();
             this.ssoStep = loginData.getLoginCredentials().getSsoStep();
             this.credentialsCompromised = context.isBreachedAccount();
+            this.username = loginData.getLoginCredentials().getUsername();
+            this.password = loginData.getLoginCredentials().getEncodedPassword();
         }
 
+        this.additionalRiskInfo = context.getAdditionalRiskInfo() != null ? context.getAdditionalRiskInfo() : null;
```
[the remark this answers is not in the document]
Comment (etrpx) [reply src/main/java/com/perimeterx/models/activities/CommonActivityDetails.java]: 👍

**Review by etrpx — COMMENTED** [review]
Thanks @ori-gold-px , please see fixes and comments

File: px_metadata.json [hunk 2]
```diff
@@ -33,6 +31,10 @@
     "custom_cookie_header",
     "sensitive_routes",
     "telemetry_command",
-    "enforcer_error"
+    "enforcer_error",
+    "pxhd",
+    "batched_activities",
+    "header_based_logger",
```
[the remark this answers is not in the document]
Comment (etrpx) [reply px_metadata.json]: 👍

**Review by ori-gold-px — APPROVED** [review]
Looks good overall. Just add excluded tests to metadata and I think it's good.

**Pull request merged**